recipes/grpc/all/test_package/conanfile.py

```diff
@@ -8,7 +8,10 @@ class TestPackageConan(ConanFile):
 
     def build_requirements(self):
         if tools.cross_building(self.settings):
-            self.build_requires(str(self.requires['protobuf']))
+            if not hasattr(self, "settings_build"):
+                return
+            else:
+                self.build_requires(str(self.requires['grpc']))
```

Comment:
I'm afraid this can fail. In a cross-building scenario this requires `gRPC` for a configuration that hasn't been built and might not be available. It ok to build-require `protobuf` because it is already available in CCI, but I have my concerns about `gRPC` here.

Reply:
Is it needed? Isn't it enough with `protobuf`?

Reply:
its needed but the package you build depends on grpc when cross compiling ..  its even needed to build itself

Reply:
because the protoc + plugin need to be buildt for the system you are compiling on in order to transpile the proto file to a cpp file that is then compiled for the target before its linked together with the grpc for the target.

Reply:
Ok, so the test will run also the grpc_cpp_plugin, not just `protoc` executable 👍